create the control panel commands

Triston-Van-Wyk opened the pull request:
I didn't do the spin it to a specific color. I also don't know if I did everything right so you can check and I can fix it if needed.

This is a fix to issue #30

Commit:
create the control panel commands

I didn't do the spin it to a sepesific color. I also don't know if I did everything right so you can check and I can fix it if needed.

devinvs commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -23,6 +23,9 @@
     public static int CLIMB_TALON = 6;
 
+    // Control panle ports
+    public static int CONTROL_PANLE_PORT = 9;
```
Typo

devinvs commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -69,6 +72,15 @@
+      public static double CONTROL_FF = 0;
+      public static double CONTROL_P = 5e-5;
+      public static double CONTROL_I = 1e-6;
+      public static double CONTROL_D = 0;
```
Leave all as zeros for now

devinvs commented on src/main/java/frc/robot/commands/ControlPanelSpin.java:
```diff
@@ -0,0 +1,52 @@
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    double control = SmartDashboard.getNumber("Control Motor Speed", 0);
```
Don't get from shuffleboard, just put 1

devinvs commented on src/main/java/frc/robot/commands/ControlPanelSpinDistance.java:
```diff
@@ -0,0 +1,120 @@
+  @Override
+  public void execute() {
+    double curVal = 0;
+    controlpanel.set(curVal);
```
This will always set it to 0...

devinvs commented on src/main/java/frc/robot/commands/ControlPanelSpinDistance.java:
```diff
@@ -0,0 +1,120 @@
+      return true;
+    }else{
+    return false;
+    }
```
I don't get the position == position... Do you know what this code does? There is a much simpler and more intuitive way to handle this. Please rewrite.

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+   * Creates a new ControlPanel.
+   */
+  public ControlPanel() {
+    control_motor = new SafeSparkMax(Constants.CONTROL_PANLE_PORT, MotorType.kBrushless);
```
Again, typo on the constant.

devinvs replied:
Also, use a safesparkmax

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+    // Always reset to factory defaults, just in case.
+    control_motor.restoreFactoryDefaults();
+
+    control_motor.clearFaults();
```
These won't be necessary if you use a safesparkmax

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+    control_pid_controller = control_motor.getPIDController();
+  }
+
+  private void configurePID() {
```
This is never called, call it in the constructor.

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+      .withPosition(0, 4)
+      .withWidget(BuiltInWidgets.kGraph)
+      .getEntry();
+  }
```
I don't want shuffleboard stuff in the subsystems any more, lets keep this in a command. Remove it please.

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+  /**This gets the velocity if the control motor
+ * @return curVal This returns the current velocity of the control panle motor 
+ */
+  public double getVelocity(){
+  double curVal = 0;
+  int curIndex =0;
+   curVal = velocityRange.get(curIndex);
+
+   return curVal;
+
+  }
```
I don't think you get what this does

devinvs replied:
Just return the velocity from the encoders.

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+
+  public void set(double control_vel){
+    control_pid_controller.setReference(control_vel, ControlType.kVelocity);
+  }
```
We want to be able to set the control type also.

devinvs commented on src/main/java/frc/robot/subsystems/ControlPanel.java:
```diff
@@ -0,0 +1,131 @@
+                                                       \/
+
+
+  This is a ice cicle it goes no where because it is only a coment. 
```
*icicle

Commit:
I updated the Control panel commands to fix the issues

I still didn't do the Issue about the code: public double getVelocity(){
double curVal = 0;
 int curIndex =0;
 curVal = velocityRange.get(curIndex);

return cur Val;

}

I didn't fix this becasue I am not sure what the encoder is.

devinvs commented on src/main/java/frc/robot/commands/ControlPanelSpinDistance.java:
```diff
@@ -38,8 +38,7 @@ public void initialize() {
   public void execute() {
-    double curVal = 0;
-    controlpanel.set(curVal);
+    controlpanel.set(0);
```
This will stop the control panel at all times... We want to it move right?

devinvs committed:
Merge branch 'develop' into feature/Contol_panle_commands

devinvs reviewed (changes requested):
See requested changes

Commit:
Update ControlPanelSpinDistance.java

I don't know if this is right. I think this may work.

Triston-Van-Wyk commented:
I fixed it so You can look over it now.
